对于新版本的一点建议

### System Info / 系統信息

xinference 0.15.0
ubuntu 22.04

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [ ] docker / docker
- [X] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

0.15.0

### The command used to start Xinference / 用以启动 xinference 的命令

xinference-local -H 0.0.0.0 -p 8000

### Reproduction / 复现过程

新版本更新以后，好像原有已注册模型不是很方便了
1. 无法编辑，点击edit以后直接白页面了，也可能是拿掉这样的操作方式了
![image](https://github.com/user-attachments/assets/8535d0b1-c2a3-449e-87dd-490816b00aba)

![image](https://github.com/user-attachments/assets/297d3630-4342-4080-8059-178171ed0f2b)

2. 因为本地模型多，包括自主量化的，和微调的，比如需要同模型不同版本，但是会发现一个问题，无法定义标题名称来区分定义，如果走自定义模型的通路，每个配置会非常复杂

浏览，如果有title来区分，可以对同模型的不同版本一目了然
![image](https://github.com/user-attachments/assets/bbed234f-3862-41ee-b150-a62ade0c8ee0)

3.这里看上去只能是默认的官方名称，少了非常多的自主性
![image](https://github.com/user-attachments/assets/db14ae78-76f4-4f46-b01d-ee0e8b27e112)

4.这是现有的自定义模型，这个对于未适配的模型倒是新增一个方便配置能力，倒是挺好的。就是把原来已适配模型的注册模式拿掉，这个会有点不方便。
![image](https://github.com/user-attachments/assets/fb4a7614-fe37-4103-b9fb-0d6391d426ca)


### Expected behavior / 期待表现


新版本，对原有的Register Model流程做了修改。
1.对于官方已适配的模型去掉了，Register环节，而是通过参数直接填写路径的方式来实现配置
2.对于官方未适配的模型，通过配置Chat template，以及stop token等方式，给用户了一个非常便捷的快速配置模型的能力，这个非常赞。
但这样会带来新问题，那就是
1.如果我需要编辑老版本已经手动注册好的模型，现在的界面是个白板，应该是被禁用了。
2.如果我有非常多的，官方已适配模型，需要做手动配置（比如不同的微调版本， 不同的自主量化版本），虽然可以通过路径的方式来快速指定，但是标题一眼看上去都是一样的，很难区分，而edit模型配置也无法做到，这一点希望官方能考虑恢复原有的方式，或者有创新的UE模式来方便这样的同模型多版本的使用场景。

新增的注册方式带来了更多的灵活性，但同时希望能考虑恢复或优化原有注册模型的方式，毕竟对于同模型的多版本管理，原有的注册模式很方便。

现在使用这个能力，只能通过老版本来实现，再用新版本推理，非常不方便。

补充一下，在新版本上，接近100个老版本注册的模型目前仅是存在于列表展现，而都无法推理使用了，只好完全回退到老版本

以上，还望慎重考虑，对注册模型模块的UI修改。

> > > @okwinds 第一个，页面无法编辑，下个版本修复
> > 
> > 
> > 赞👍🏻
> > 后来，我又想了想，目前用xinference一个模型大概，gguf，gptq，awq，tensor-compress，fp8，fp4，这些量化方法都会注册，加上不同量化方法的不同量化等级，然后再加上不同项目的微调模型的不同方法的不同量化等级，也许仅一个模型就会有30个左右的版本。原有的注册方式好处就是注册完毕，custom列表里就能很方便的找到，也方便管理，如果仅用目前官方适配的模型入口通道，虽然简单，但是每次都要换不同的路径，以及不同的推理参数，反而会有点麻烦。。。
> > 如果后面能做兼容，那就太棒了 👍🏻
> 
> 首先会修页面的问题。 然后关于你的注册，我们下个版本页面和接口会兼容只写model_family的情况。其实后端接口已经兼容以前的注册方式，但是不知道为啥你拉不起来，上面我写的示例是ok的。页面会兼容这种方式。

嗯，是老版本上注册过的，新版本可以拉起来，但是只要对话就报错，比如下面这段：

```
2024-09-09 16:20:26,376 uvicorn.access 3673995 INFO     127.0.0.1:53534 - "POST /v1/chat/completions HTTP/1.1" 200
2024-09-09 16:20:26,377 xinference.core.model 3679441 ERROR    [request 5e0328ba-6e84-11ef-8b5b-00155d168ff2] Leave chat, error: , elapsed time: 0 s
Traceback (most recent call last):
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
    response = await self._call_wrapper_json(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
    return await self._call_wrapper("json", fn, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
    return await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
    ret = await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
    assert self.model_family.chat_template is not None
AssertionError
2024-09-09 16:20:26,379 xinference.api.restful_api 3673995 ERROR    Chat completion stream got an error: [address=0.0.0.0:44631, pid=3679441] 
Traceback (most recent call last):
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1688, in stream_results
    iterator = await model.chat(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 96, in wrapped_func
    ret = await fn(self, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 462, in _wrapper
    r = await func(self, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
    response = await self._call_wrapper_json(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
    return await self._call_wrapper("json", fn, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
    return await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
    ret = await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
    assert self.model_family.chat_template is not None
AssertionError: [address=0.0.0.0:44631, pid=3679441] 
2024-09-09 16:20:26,428 uvicorn.access 3673995 INFO     172.25.16.1:60661 - "POST /Qwen2-7B-Instruct/run/predict HTTP/1.1" 200
```

> > > > @okwinds 第一个，页面无法编辑，下个版本修复
> > > 
> > > 
> > > 赞👍🏻
> > > 后来，我又想了想，目前用xinference一个模型大概，gguf，gptq，awq，tensor-compress，fp8，fp4，这些量化方法都会注册，加上不同量化方法的不同量化等级，然后再加上不同项目的微调模型的不同方法的不同量化等级，也许仅一个模型就会有30个左右的版本。原有的注册方式好处就是注册完毕，custom列表里就能很方便的找到，也方便管理，如果仅用目前官方适配的模型入口通道，虽然简单，但是每次都要换不同的路径，以及不同的推理参数，反而会有点麻烦。。。
> > > 如果后面能做兼容，那就太棒了 👍🏻
> > 
> > 
> > 首先会修页面的问题。 然后关于你的注册，我们下个版本页面和接口会兼容只写model_family的情况。其实后端接口已经兼容以前的注册方式，但是不知道为啥你拉不起来，上面我写的示例是ok的。页面会兼容这种方式。
> 
> 嗯，是老版本上注册过的，新版本可以拉起来，但是只要对话就报错，比如下面这段：
> 
> ```
> 2024-09-09 16:20:26,376 uvicorn.access 3673995 INFO     127.0.0.1:53534 - "POST /v1/chat/completions HTTP/1.1" 200
> 2024-09-09 16:20:26,377 xinference.core.model 3679441 ERROR    [request 5e0328ba-6e84-11ef-8b5b-00155d168ff2] Leave chat, error: , elapsed time: 0 s
> Traceback (most recent call last):
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
>     ret = await func(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
>     response = await self._call_wrapper_json(
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
>     return await self._call_wrapper("json", fn, *args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
>     return await fn(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
>     ret = await fn(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
>     assert self.model_family.chat_template is not None
> AssertionError
> 2024-09-09 16:20:26,379 xinference.api.restful_api 3673995 ERROR    Chat completion stream got an error: [address=0.0.0.0:44631, pid=3679441] 
> Traceback (most recent call last):
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1688, in stream_results
>     iterator = await model.chat(
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
>     return self._process_result_message(result)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
>     raise message.as_instanceof_cause()
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
>     result = await self._run_coro(message.message_id, coro)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
>     return await coro
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
>     return await super().__on_receive__(message)  # type: ignore
>   File "xoscar/core.pyx", line 558, in __on_receive__
>     raise ex
>   File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
>     async with self._lock:
>   File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
>     with debug_async_timeout('actor_lock_timeout',
>   File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
>     result = await result
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 96, in wrapped_func
>     ret = await fn(self, *args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 462, in _wrapper
>     r = await func(self, *args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
>     ret = await func(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
>     response = await self._call_wrapper_json(
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
>     return await self._call_wrapper("json", fn, *args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
>     return await fn(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
>     ret = await fn(*args, **kwargs)
>   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
>     assert self.model_family.chat_template is not None
> AssertionError: [address=0.0.0.0:44631, pid=3679441] 
> 2024-09-09 16:20:26,428 uvicorn.access 3673995 INFO     172.25.16.1:60661 - "POST /Qwen2-7B-Instruct/run/predict HTTP/1.1" 200
> ```

我这边无法复现。你贴一个你报错这个的对应的json给我，我试试。

FYI，看看能复现不

`{"version": 1, "context_length": 32768, "model_name": "Qwen2-7B-Instruct", "model_lang": ["en", "zh"], "model_ability": ["generate", "chat"], "model_description": "/home/gavin/llm/Qwen2/Qwen2-7B-Instruct", "model_family": "qwen2-instruct", "model_specs": [{"model_format": "pytorch", "model_size_in_billions": 7, "quantizations": ["4-bit", "8-bit", "none"], "model_id": null, "model_hub": "huggingface", "model_uri": "/home/gavin/llm/Qwen2/Qwen2-7B-Instruct", "model_revision": null}], "prompt_style": {"style_name": "QWEN", "system_prompt": "You are a helpful assistant.", "roles": ["user", "assistant"], "intra_message_sep": "\n", "inter_message_sep": "", "stop": ["<|endoftext|>", "<|im_start|>", "<|im_end|>"], "stop_token_ids": [151643, 151644, 151645]}}`

> > > > > @okwinds 第一个，页面无法编辑，下个版本修复
> > > > 
> > > > 
> > > > 赞👍🏻
> > > > 后来，我又想了想，目前用xinference一个模型大概，gguf，gptq，awq，tensor-compress，fp8，fp4，这些量化方法都会注册，加上不同量化方法的不同量化等级，然后再加上不同项目的微调模型的不同方法的不同量化等级，也许仅一个模型就会有30个左右的版本。原有的注册方式好处就是注册完毕，custom列表里就能很方便的找到，也方便管理，如果仅用目前官方适配的模型入口通道，虽然简单，但是每次都要换不同的路径，以及不同的推理参数，反而会有点麻烦。。。
> > > > 如果后面能做兼容，那就太棒了 👍🏻
> > > 
> > > 
> > > 首先会修页面的问题。 然后关于你的注册，我们下个版本页面和接口会兼容只写model_family的情况。其实后端接口已经兼容以前的注册方式，但是不知道为啥你拉不起来，上面我写的示例是ok的。页面会兼容这种方式。
> > 
> > 
> > 嗯，是老版本上注册过的，新版本可以拉起来，但是只要对话就报错，比如下面这段：
> > ```
> > 2024-09-09 16:20:26,376 uvicorn.access 3673995 INFO     127.0.0.1:53534 - "POST /v1/chat/completions HTTP/1.1" 200
> > 2024-09-09 16:20:26,377 xinference.core.model 3679441 ERROR    [request 5e0328ba-6e84-11ef-8b5b-00155d168ff2] Leave chat, error: , elapsed time: 0 s
> > Traceback (most recent call last):
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
> >     ret = await func(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
> >     response = await self._call_wrapper_json(
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
> >     return await self._call_wrapper("json", fn, *args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
> >     return await fn(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
> >     ret = await fn(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
> >     assert self.model_family.chat_template is not None
> > AssertionError
> > 2024-09-09 16:20:26,379 xinference.api.restful_api 3673995 ERROR    Chat completion stream got an error: [address=0.0.0.0:44631, pid=3679441] 
> > Traceback (most recent call last):
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1688, in stream_results
> >     iterator = await model.chat(
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
> >     return self._process_result_message(result)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
> >     raise message.as_instanceof_cause()
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
> >     result = await self._run_coro(message.message_id, coro)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
> >     return await coro
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
> >     return await super().__on_receive__(message)  # type: ignore
> >   File "xoscar/core.pyx", line 558, in __on_receive__
> >     raise ex
> >   File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
> >     async with self._lock:
> >   File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
> >     with debug_async_timeout('actor_lock_timeout',
> >   File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
> >     result = await result
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 96, in wrapped_func
> >     ret = await fn(self, *args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 462, in _wrapper
> >     r = await func(self, *args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
> >     ret = await func(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
> >     response = await self._call_wrapper_json(
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
> >     return await self._call_wrapper("json", fn, *args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
> >     return await fn(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
> >     ret = await fn(*args, **kwargs)
> >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
> >     assert self.model_family.chat_template is not None
> > AssertionError: [address=0.0.0.0:44631, pid=3679441] 
> > 2024-09-09 16:20:26,428 uvicorn.access 3673995 INFO     172.25.16.1:60661 - "POST /Qwen2-7B-Instruct/run/predict HTTP/1.1" 200
> > ```
> 
> 我这边无法复现。你贴一个你报错这个的对应的json给我，我试试。

> FYI，看看能复现不
> 
> ```
> {"version": 1, "context_length": 32768, "model_name": "Qwen2-7B-Instruct", "model_lang": ["en", "zh"], "model_ability": ["generate", "chat"], "model_description": "/home/gavin/llm/Qwen2/Qwen2-7B-Instruct", "model_family": "qwen2-instruct", "model_specs": [{"model_format": "pytorch", "model_size_in_billions": 7, "quantizations": ["4-bit", "8-bit", "none"], "model_id": null, "model_hub": "huggingface", "model_uri": "/home/gavin/llm/Qwen2/Qwen2-7B-Instruct", "model_revision": null}], "prompt_style": {"style_name": "QWEN", "system_prompt": "You are a helpful assistant.", "roles": ["user", "assistant"], "intra_message_sep": "\n", "inter_message_sep": "", "stop": ["<|endoftext|>", "<|im_start|>", "<|im_end|>"], "stop_token_ids": [151643, 151644, 151645]}}
> ```
> 
> > > > > > @okwinds 第一个，页面无法编辑，下个版本修复
> > > > > 
> > > > > 
> > > > > 赞👍🏻
> > > > > 后来，我又想了想，目前用xinference一个模型大概，gguf，gptq，awq，tensor-compress，fp8，fp4，这些量化方法都会注册，加上不同量化方法的不同量化等级，然后再加上不同项目的微调模型的不同方法的不同量化等级，也许仅一个模型就会有30个左右的版本。原有的注册方式好处就是注册完毕，custom列表里就能很方便的找到，也方便管理，如果仅用目前官方适配的模型入口通道，虽然简单，但是每次都要换不同的路径，以及不同的推理参数，反而会有点麻烦。。。
> > > > > 如果后面能做兼容，那就太棒了 👍🏻
> > > > 
> > > > 
> > > > 首先会修页面的问题。 然后关于你的注册，我们下个版本页面和接口会兼容只写model_family的情况。其实后端接口已经兼容以前的注册方式，但是不知道为啥你拉不起来，上面我写的示例是ok的。页面会兼容这种方式。
> > > 
> > > 
> > > 嗯，是老版本上注册过的，新版本可以拉起来，但是只要对话就报错，比如下面这段：
> > > ```
> > > 2024-09-09 16:20:26,376 uvicorn.access 3673995 INFO     127.0.0.1:53534 - "POST /v1/chat/completions HTTP/1.1" 200
> > > 2024-09-09 16:20:26,377 xinference.core.model 3679441 ERROR    [request 5e0328ba-6e84-11ef-8b5b-00155d168ff2] Leave chat, error: , elapsed time: 0 s
> > > Traceback (most recent call last):
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
> > >     ret = await func(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
> > >     response = await self._call_wrapper_json(
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
> > >     return await self._call_wrapper("json", fn, *args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
> > >     return await fn(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
> > >     ret = await fn(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
> > >     assert self.model_family.chat_template is not None
> > > AssertionError
> > > 2024-09-09 16:20:26,379 xinference.api.restful_api 3673995 ERROR    Chat completion stream got an error: [address=0.0.0.0:44631, pid=3679441] 
> > > Traceback (most recent call last):
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1688, in stream_results
> > >     iterator = await model.chat(
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
> > >     return self._process_result_message(result)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
> > >     raise message.as_instanceof_cause()
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
> > >     result = await self._run_coro(message.message_id, coro)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
> > >     return await coro
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
> > >     return await super().__on_receive__(message)  # type: ignore
> > >   File "xoscar/core.pyx", line 558, in __on_receive__
> > >     raise ex
> > >   File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
> > >     async with self._lock:
> > >   File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
> > >     with debug_async_timeout('actor_lock_timeout',
> > >   File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
> > >     result = await result
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 96, in wrapped_func
> > >     ret = await fn(self, *args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 462, in _wrapper
> > >     r = await func(self, *args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
> > >     ret = await func(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
> > >     response = await self._call_wrapper_json(
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
> > >     return await self._call_wrapper("json", fn, *args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
> > >     return await fn(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
> > >     ret = await fn(*args, **kwargs)
> > >   File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 666, in async_chat
> > >     assert self.model_family.chat_template is not None
> > > AssertionError: [address=0.0.0.0:44631, pid=3679441] 
> > > 2024-09-09 16:20:26,428 uvicorn.access 3673995 INFO     172.25.16.1:60661 - "POST /Qwen2-7B-Instruct/run/predict HTTP/1.1" 200
> > > ```
> > 
> > 
> > 我这边无法复现。你贴一个你报错这个的对应的json给我，我试试。

我改成了自己的路径:
```
{
    "version": 1,
    "context_length": 32768,
    "model_name": "Qwen2-7B-Instruct-test",
    "model_lang": [
        "en",
        "zh"
    ],
    "model_ability": [
        "generate",
        "chat"
    ],
    "model_description": "/home/gavin/llm/Qwen2/Qwen2-7B-Instruct",
    "model_family": "qwen2-instruct",
    "model_specs": [
        {
            "model_format": "pytorch",
            "model_size_in_billions": 7,
            "quantizations": [
                "4-bit",
                "8-bit",
                "none"
            ],
            "model_uri": "/new_data2/lichengjie/doc-insights/cache/cache/qwen1.5-chat-pytorch-0_5b"
        }
    ],
    "prompt_style": {
        "style_name": "QWEN",
        "system_prompt": "You are a helpful assistant.",
        "roles": [
            "user",
            "assistant"
        ],
        "intra_message_sep": "\n",
        "inter_message_sep": "",
        "stop": [
            "<|endoftext|>",
            "<|im_start|>",
            "<|im_end|>"
        ],
        "stop_token_ids": [
            151643,
            151644,
            151645
        ]
    }
}
```
vllm部署，网页对话完全没问题。
其实代码里是，model_family写内置模型，我会把相应配置替换。所以只要model_family写对就行了。你这个是没问题的。

> @okwinds #2268 修复了前端白屏的问题。 现在理应你的以前的注册方式前端后端都没问题了。你可以试下main上的最新代码。有问题再在这里贴出来。

我今天更新的最新的版本0.15.1，并通过conda create重建新版本环境，还是以前配置过的模型，启动的时候报错，如下：

```
2024-09-14 22:13:24,898 vllm.worker.model_runner 38695 INFO     Loading model weights took 14.2487 GB
2024-09-14 22:13:25,416 vllm.executor.gpu_executor 38695 INFO     # GPU blocks: 6398, # CPU blocks: 4681
2024-09-14 22:13:25,609 vllm.worker.model_runner 38695 INFO     Capturing the model for CUDA graphs. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI.
2024-09-14 22:13:25,610 vllm.worker.model_runner 38695 INFO     CUDA graphs can take additional 1~3 GiB memory per GPU. If you are running out of memory, consider decreasing `gpu_memory_utilization` or enforcing eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.
2024-09-14 22:13:31,582 vllm.worker.model_runner 38695 INFO     Graph capturing finished in 6 secs.
2024-09-14 22:13:31,593 xinference.core.worker 38163 INFO     [request 7ec446ec-72a3-11ef-9e40-00155d89527b] Leave launch_builtin_model, elapsed time: 11 s
2024-09-14 22:13:42,451 xinference.core.model 38695 ERROR    [request 8bec1192-72a3-11ef-afd0-00155d89527b] Leave chat, error: , elapsed time: 0 s
Traceback (most recent call last):
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
    response = await self._call_wrapper_json(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
    return await self._call_wrapper("json", fn, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
    return await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
    ret = await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 674, in async_chat
    assert self.model_family.chat_template is not None
AssertionError
2024-09-14 22:13:42,453 xinference.api.restful_api 38117 ERROR    Chat completion stream got an error: [address=0.0.0.0:36555, pid=38695]
Traceback (most recent call last):
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1891, in stream_results
    iterator = await model.chat(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 96, in wrapped_func
    ret = await fn(self, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xoscar/api.py", line 462, in _wrapper
    r = await func(self, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 570, in chat
    response = await self._call_wrapper_json(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 407, in _call_wrapper_json
    return await self._call_wrapper("json", fn, *args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 120, in _async_wrapper
    return await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/model.py", line 416, in _call_wrapper
    ret = await fn(*args, **kwargs)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/model/llm/vllm/core.py", line 674, in async_chat
    assert self.model_family.chat_template is not None
AssertionError: [address=0.0.0.0:36555, pid=38695]
Traceback (most recent call last):
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/queueing.py", line 536, in process_events
    response = await route_utils.call_process_api(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/route_utils.py", line 322, in call_process_api
    output = await app.get_blocks().process_api(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/blocks.py", line 1935, in process_api
    result = await self.call_function(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/blocks.py", line 1532, in call_function
    prediction = await utils.async_iteration(iterator)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/utils.py", line 671, in async_iteration
    return await iterator.__anext__()
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/utils.py", line 776, in asyncgen_wrapper
    response = await iterator.__anext__()
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/chat_interface.py", line 653, in _stream_fn
    first_response = await async_iteration(generator)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/utils.py", line 671, in async_iteration
    return await iterator.__anext__()
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/utils.py", line 664, in __anext__
    return await anyio.to_thread.run_sync(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 2177, in run_sync_in_worker_thread
    return await future
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 859, in run
    result = context.run(func, *args)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/gradio/utils.py", line 647, in run_sync_iterator_async
    return next(iterator)
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/core/chat_interface.py", line 122, in generate_wrapper
    for chunk in model.chat(
  File "/home/gavin/miniconda3/envs/xinf_cp310/lib/python3.10/site-packages/xinference/client/common.py", line 51, in streaming_response_iterator
    raise Exception(str(error))
Exception: [address=0.0.0.0:36555, pid=38695]
```

--- a/xinference/model/llm/__init__.py
+++ b/xinference/model/llm/__init__.py
@@ -121,7 +121,7 @@ def register_custom_model():
                 with codecs.open(
                     os.path.join(user_defined_llm_dir, f), encoding="utf-8"
                 ) as fd:
-                    user_defined_llm_family = CustomLLMFamilyV1.parse_obj(json.load(fd))
+                    user_defined_llm_family = CustomLLMFamilyV1.parse_raw(fd.read())
                     register_llm(user_defined_llm_family, persist=False)
             except Exception as e:
                 warnings.warn(f"{user_defined_llm_dir}/{f} has error, {e}")

